### PR TITLE
Add progress bar to pandas apply function

### PR DESCRIPTION
**Pull request by hadim**
Tell me if you are ok to include this.

### CODE REVIEW

**Comment by hadim** [comment]
The following include a big copy-paste code from `_tqdm.py`. I really need to find a way to avoid this code duplication.

**Comment by kmike** [comment]
I like this feature (wanted a tqdm for DataFrame.apply myself). 
+1 to find a way to reduce code duplication. 
It'd be also nice to mention this feature in README.

**Comment by hadim** [comment]
Cool !

I will work on it next week.

File: tqdm/tests/tests_pandas.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+from nose.plugins.skip import SkipTest
+from nose.tools import with_setup
+
+
+def setup_pandas():
+    try:
+        from tqdm import enable_progress_apply
+        enable_progress_apply()
+    except:
+        raise SkipTest
```
Comment (kmike) [inline tqdm/tests/tests_pandas.py]: To run this test on Travis pandas should be added to tox.ini

**Comment by lrq3000** [comment]
This should be updated with the new object oriented architecture of tqdm, and merged! This is a great addition.

**Comment by hadim** [comment]
Sure but I don't have time right now to dig into this. Maybe later !

**Comment by lrq3000** [comment]
Yes I understand, it's not urgent, but it sure is a very good added value :D I will myself use that in the future without a doubt

**Comment by casperdcl** [comment]
Having never used `pandas` due to its various inefficiencies, I'm not sure if this is what you guys wanted. But I tried...

**Comment by casperdcl** [comment]
I don't know if you'd like me to revert back to `tqdm_pandas(tqdm(**kwargs))` instead of `tqdm_pandas(**kwargs)` since the former allows for `tqdm_gui` and other derived classes to be used instead...

**Comment by lrq3000** [comment]
@casperdcl : `tqdm_pandas(**kwargs)` is simpler, but since we started to modularize the code, I think it would be more logical to revert to `tqdm_pandas(tqdm(**kwargs))`, so that future derived classes will be useable without breaking compatibility. Thank's for the work :+1: 

I don't use pandas currently, can someone that use pandas extensively test this out?

**Comment by casperdcl** [comment]
The rewritten tests seem to work... Travis seems to be taking its time to install `numpy` and `pandas`, though.

**Comment by casperdcl** [comment]
Also todo: answer the following to gain popularity...

- [ ] http://stackoverflow.com/questions/18603270/progress-indicator-during-pandas-operations-python
- [ ] http://stackoverflow.com/questions/24737683/how-do-i-create-a-progress-bar-when-a-dataframe-is-initializing

**Comment by casperdcl** [comment]
@hadim @kmike can you check if this PR makes your day ;) ?

**Comment by hadim** [comment]
Will do this week.

**Comment by hadim** [comment]
Working nicely to me. Very nice work @casperdcl !

Good to merge for me.

**Comment by casperdcl** [comment]
required surprisingly few loc, actually... kind of like the rest of `tqdm`. Most of our repo is documentation and tests :) ... it's really a useful template for other projects, hah

**Comment by lrq3000** [comment]
Totally agree! :+1:

2015-12-01 0:55 GMT+01:00 Casper da Costa-Luis <notifications@github.com>:

**Comment by manugarri** [comment]
This is way cool, looking forward to see it on a release!

**Comment by casperdcl** [comment]
@manugarri it's been released for ages... try `pip install -U tqdm` if you need to upgrade... Or use the version from github.

**Pull request merged**